src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { getUserById } from '../../services/getUserById';
+import usersFromServer from '../../api/users';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
```

Comment:
I'm not sure that you need state for title and id error.
You can check it like that.

```
  const isTitleError = !title;
  const isUserError = !userId;
```
And than you can handle id change like that
```
   <select
            data-cy="userSelect"
            value={userId}
            onChange={event => setUserId(event.target.value)}
          >
          ```
          instead of handleUserIdChange

Reply:
> const isTitleError = !title;

If you do ` const isTitleError = !title;` a user will see an error message even they did nothing, so it's ok